src-test/src/com/etendoerp/etendorx/datasource/EntityFieldSelectedFiltersTest.java

```diff
@@ -0,0 +1,256 @@
+
+/**
+ * Test class for {@code EntityFieldSelectedFilters} in {@link ManageEntityFieldsDS}.
+ * <p>
+ * This class ensures the proper functionality of the filter-related features,
+ * including collection initialization, additions, modifications, and validation of properties.
+ * </p>
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class EntityFieldSelectedFiltersTest {
+
+  private Object filters;
+  private Class<?> filtersClass;
+
+  /**
+   * Sets up the test environment by initializing the {@code EntityFieldSelectedFilters} class
+   * using reflection before each test method.
+   *
+   * @throws Exception if any setup error occurs
+   */
+  @Before
+  public void setUp() throws Exception {
+    filtersClass = Class.forName("com.etendoerp.etendorx.datasource.ManageEntityFieldsDS$EntityFieldSelectedFilters");
+    Constructor<?> constructor = filtersClass.getDeclaredConstructor(ManageEntityFieldsDS.class);
+    constructor.setAccessible(true);
+    filters = constructor.newInstance(new ManageEntityFieldsDS());
+  }
+
+  /**
+   * Tests that all collections within {@code EntityFieldSelectedFilters} are properly initialized.
+   *
+   * @throws Exception if any reflection-related error occurs
+   */
+  @Test
+  public void testCollectionInitialization() throws Exception {
+    assertNotNull("SelectedMappingValues should be initialized",
+        filtersClass.getMethod(TestUtils.GET_SELECTED_MAPPING_VALUES).invoke(filters));
+    assertNotNull("SelectedIds should be initialized",
+        filtersClass.getMethod("getSelectedIds").invoke(filters));
+    assertNotNull("ModuleIds should be initialized",
+        filtersClass.getMethod("getModuleIds").invoke(filters));
+    assertNotNull("FieldMappingIds should be initialized",
+        filtersClass.getMethod("getFieldMappingIds").invoke(filters));
+    assertNotNull("JavaMappingIds should be initialized",
+        filtersClass.getMethod("getJavaMappingIds").invoke(filters));
+  }
+
+  /**
+   * Tests the addition of multiple elements to various collections in {@code EntityFieldSelectedFilters}.
+   *
+   * @throws Exception if any reflection-related error occurs
+   */
+  @Test
+  public void testMultipleAdditionsToCollections() throws Exception {
+    List<String> testIds = Arrays.asList("id1", "id2", "id3", "id4", "id5");
+
+    for (String id : testIds) {
+      filtersClass.getMethod("addSelectedID", String.class).invoke(filters, id);
+      filtersClass.getMethod("addModuleIds", String.class).invoke(filters, id);
+      filtersClass.getMethod("addJavaMappingIds", String.class).invoke(filters, id);
+      filtersClass.getMethod("addFieldMappingValues", String.class).invoke(filters, id);
+      filtersClass.getMethod("addEtrxProjectionEntityRelatedIds", String.class).invoke(filters, id);
+      filtersClass.getMethod("addEtrxConstantValueIds", String.class).invoke(filters, id);
+    }
+
+    for (String methodName : Arrays.asList(
+        "getSelectedIds", "getModuleIds", "getJavaMappingIds",
```

Comment:
`getJavaMappingIds`
I recommend passing to utils as they are used several times